pkg/provisioner/ironic/ironic.go

```diff
@@ -321,6 +321,53 @@ func (p *ironicProvisioner) getNode() (*nodes.Node, error) {
 	}
 }
 
+// Verifies that node has port assigned by Ironic.
+func (p *ironicProvisioner) nodeHasAssignedPort(ironicNode *nodes.Node) (bool, error) {
+	opts := ports.ListOpts{
+		Fields:   []string{"node_uuid"},
+		NodeUUID: ironicNode.UUID,
+	}
+
+	pager := ports.List(p.client, opts)
+	if pager.Err != nil {
```

Comment:
This can't actually happen, because no request has been made yet.

Reply:
code was modeled after `listAllPorts()`
https://github.com/metal3-io/baremetal-operator/blob/2e6cac3facf3ebc913f42084390780954dec70bb/pkg/provisioner/ironic/ironic.go#L290

so I assume that `listAllPorts()` also incorrect?

Reply:
> so I assume that `listAllPorts()` also incorrect?

Indeed.